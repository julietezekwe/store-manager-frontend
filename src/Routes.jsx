import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Login from './components/common/Login.jsx';
import ProductList from './components/common/ProductList.jsx';
import AllUsers from './components/Admin/AllUsers.jsx';
import AllSales from './components/Admin/AllSales.jsx';
import ProductDetails from './components/common/ProductDetails.jsx';
import AdminRoutes from './AdminRoutes.jsx';
import AttendantRoutes from './AttendantRoutes.jsx';
import CommonRoutes from './CommonRoutes.jsx';

const Routes = () => (
  <Switch >
    <Route exact path="/" default component={Login} />
    <CommonRoutes
      exact path="/products/:id"
      component={ProductDetails} />
    <AdminRoutes exact
      path="/admin" component={ProductList} />
    <AdminRoutes
      exact path="/admin/users" component={AllUsers} />
    <AdminRoutes
      exact path="/admin/sales" component={AllSales} />
    <CommonRoutes
      exact path="/products/:id"
      component={ProductDetails} />
    <AttendantRoutes exact
      path="/attendant" component={ProductList} />
  </Switch>
);

export default Routes;
