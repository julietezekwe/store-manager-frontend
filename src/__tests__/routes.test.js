/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import Routes from '../Routes.jsx';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Routes />);
});
describe('Routes Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
