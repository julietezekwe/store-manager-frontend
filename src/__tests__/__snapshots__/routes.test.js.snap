// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component should match snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    default={true}
    exact={true}
    path="/"
  />
  <Connect(AdminRoutes)
    component={[Function]}
    exact={true}
    path="/admin"
  />
  <Connect(AdminRoutes)
    component={[Function]}
    exact={true}
    path="/admin/users"
  />
  <Connect(AdminRoutes)
    component={[Function]}
    exact={true}
    path="/admin/sales"
  />
  <Connect(AttendantRoutes)
    component={[Function]}
    exact={true}
    path="/attendant"
  />
</Switch>
`;
